mypy/constraints.py

```diff
@@ -373,11 +373,12 @@ def visit_instance(self, template: Instance) -> List[Constraint]:
                 cb = infer_constraints(template.args[0], item, SUPERTYPE_OF)
                 res.extend(cb)
             return res
+        elif isinstance(actual, TupleType) and self.direction == SUPERTYPE_OF:
+            return infer_constraints(template, actual.fallback, self.direction)
         elif (isinstance(actual, TupleType) and template.type.is_protocol and
```

Comment:
I don't see how this branch can ever be entered - when this condition is true, the condition of the branch above will be true, and this won't get entered.

Reply:
Yes, this branch is redundant now.